op-chain-ops/genesis/config.go

```diff
@@ -18,50 +19,247 @@ import (
+}
+
+type deployConfigJSON struct {
+	L1StartingBlockTag               json.RawMessage `json:"l1StartingBlockTag"`
+	L1ChainID                        int64           `json:"l1ChainID"`
+	L2ChainID                        int64           `json:"l2ChainID"`
+	L2BlockTime                      uint            `json:"l2BlockTime"`
+	MaxSequencerDrift                uint            `json:"maxSequencerDrift"`
+	SequencerWindowSize              uint            `json:"sequencerWindowSize"`
+	ChannelTimeout                   uint            `json:"channelTimeout"`
+	P2PSequencerAddress              common.Address  `json:"p2pSequencerAddress"`
+	OptimismL2FeeRecipient           common.Address  `json:"optimismL2FeeRecipient"`
+	BatchInboxAddress                common.Address  `json:"batchInboxAddress"`
+	BatchSenderAddress               common.Address  `json:"batchSenderAddress"`
+	L2OutputOracleSubmissionInterval uint            `json:"l2OutputOracleSubmissionInterval"`
+	L2OutputOracleStartingTimestamp  int             `json:"l2OutputOracleStartingTimestamp"`
+	L2OutputOracleProposer           common.Address  `json:"l2OutputOracleProposer"`
+	L2OutputOracleOwner              common.Address  `json:"l2OutputOracleOwner"`
+	L1BlockTime                      uint64          `json:"l1BlockTime"`
+	CliqueSignerAddress              common.Address  `json:"cliqueSignerAddress"`
+	OptimismBaseFeeRecipient         common.Address  `json:"optimismBaseFeeRecipient"`
+	OptimismL1FeeRecipient           common.Address  `json:"optimismL1FeeRecipient"`
+	GasPriceOracleOwner              common.Address  `json:"gasPriceOracleOwner"`
+	GasPriceOracleOverhead           uint            `json:"gasPriceOracleOverhead"`
+	GasPriceOracleScalar             uint            `json:"gasPriceOracleScalar"`
+	GasPriceOracleDecimals           uint            `json:"gasPriceOracleDecimals"`
+	L2CrossDomainMessengerOwner      common.Address  `json:"l2CrossDomainMessengerOwner"`
+	L2GenesisBlockNonce              hexutil.Uint64  `json:"l2GenesisBlockNonce"`
+	L2GenesisBlockExtraData          hexutil.Bytes   `json:"l2GenesisBlockExtraData"`
+	L2GenesisBlockGasLimit           hexutil.Uint64  `json:"l2GenesisBlockGasLimit"`
+	L2GenesisBlockDifficulty         string          `json:"l2GenesisBlockDifficulty"`
+	L2GenesisBlockMixHash            common.Hash     `json:"l2GenesisBlockMixHash"`
+	L2GenesisBlockCoinbase           common.Address  `json:"l2GenesisBlockCoinbase"`
+	L2GenesisBlockNumber             hexutil.Uint64  `json:"l2GenesisBlockNumber"`
+	L2GenesisBlockGasUsed            hexutil.Uint64  `json:"l2GenesisBlockGasUsed"`
+	L2GenesisBlockParentHash         common.Hash     `json:"l2GenesisBlockParentHash"`
+	L2GenesisBlockBaseFeePerGas      string          `json:"l2GenesisBlockBaseFeePerGas"`
+	L1GenesisBlockTimestamp          hexutil.Uint64  `json:"l1GenesisBlockTimestamp"`
+	L1GenesisBlockNonce              hexutil.Uint64  `json:"l1GenesisBlockNonce"`
```

Comment:
It's a lot of fields to maintain 2 separate structs, I wouldn't be opposed to using hexutil types in the main struct if it lets us only maintain 1 struct. What do you think?

Reply:
IMO I'd rather have a bunch of additional fields than leak details about how the struct is serialized. We either maintain lots of fields or force callers to type cast/call unnecessary methods everywhere this struct is used, and I prefer the former.

The nice thing about the test is that it'll complain if the structs fall out of sync.